lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -134,46 +114,27 @@ defmodule ExUnit.Server do
       %{config | sync_cases: Set.put(config.sync_cases, name)}}
   end
 
-  def handle_cast(request, config) do
-    super(request, config)
-  end
-
   def handle_info({:DOWN, ref, _, _, _}, config) do
-    refs = Map.delete(config.log_captures, ref)
-    maybe_add_console(refs, config.log_status)
-    remove_capture(ref)
-    {:noreply, %{config | log_captures: refs}}
-  end
-
-  def handle_info({:gen_event_EXIT, _, :normal}, config) do
+    config = remove_log_capture(ref, config)
     {:noreply, config}
   end
 
-  def handle_info({:gen_event_EXIT, {Console, ref}, reason}, config) do
-    refs = Map.put(config.log_captures, ref, reason)
-    maybe_add_console(refs, config.log_status)
-    {:noreply, %{config | log_captures: refs}}
+  def handle_info(_msg, state) do
+    {:noreply, state}
   end
 
-  def handle_info(msg, state) do
-    super(msg, state)
-  end
-
-  defp remove_capture(ref) do
-    case GenEvent.remove_handler(Logger, {Console, ref}, nil) do
-      {:error, :not_found} ->
-        receive do
-          {:gen_event_EXIT, {Console, ^ref}, reason} ->
-            {:error, reason}
-        after
-          0 -> {:error, :not_found}
-        end
-      :ok -> :ok
+  defp remove_log_capture(ref, %{log_captures: refs} = config) do
+    if ref in refs do
+      refs = HashSet.delete(refs, ref)
+      maybe_add_console(refs, config.log_status)
+      %{config | log_status: nil, log_captures: refs}
```

Comment:
I think we shouldn't change `log_status` to `nil` as it won't match in `maybe_add_console/2` next time.